src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -492,13 +495,26 @@ const WorkflowView = _.flow(
     return this.isSingle() || !!rootEntityType
   }
 
+  loadNewMethodConfig = _.flow(
+    withErrorReporting('Error updating config'),
+    Utils.withBusyState(v => this.setState({ updatingConfig: v }))
+  )(async newSnapshotId => {
+    const { ajax: { Methods } } = this.props
+    const { modifiedConfig: { methodRepoMethod: { methodNamespace, methodName } } } = this.state
+    const config = await Methods.template({ methodNamespace, methodName, methodVersion: newSnapshotId })
+    const modifiedInputsOutputs = await Methods.configInputsOutputs(config)
+    this.setState({ modifiedInputsOutputs: _.update('inputs', _.sortBy('optional'), modifiedInputsOutputs) })
```

Comment:
This is, I think, the 3rd place where this manipulation exists. Probably worth extracting a function for it.